bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorDefaultConnector.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.epsonprojector.internal.connector;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorException;
+
+/**
+ * Default Connector
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+@NonNullByDefault
+public class EpsonProjectorDefaultConnector implements EpsonProjectorConnector {
```

Comment:
You could introduce two projector things, one for serial port communication and another for ip communication, where then thing description files contain mandatory parameters for thing (serial port in another thing and ip address + port in another). Framework then force user to define all necessary parameters for thing and this class could be then removed.

Reply:
I tried to do that inside the handler factory and create/pass the appropriate device object to the handler based on the which thing is used.... But the configuration is not available there so I would have to refactor the device class and the connector classes to take configuration via a setter among other things.

Reply:
You don't need to do any special changes to factory or handler itself. Both things can use same configuration class and handler internally.

I have implemented similar to other bindings as well. Take a look e.g. rfxcom or nibe heat pump binding. I'm sure there is other bindings as well, which support multiple communication techonogies.

Reply:
Thanks, I made separate things per connection type.